src/hotspot/share/runtime/os.cpp

```diff
@@ -1528,6 +1528,53 @@ bool os::set_boot_path(char fileSep, char pathSep) {
   return false;
 }
 
+static char* _image_release_file_content = nullptr;
+
+void os::read_image_release_file() {
+  const char* home = Arguments::get_java_home();
+  stringStream ss;
+  ss.print("%s/release", home);
+
+  if (_image_release_file_content == nullptr) {
+    FILE* file = fopen(ss.base(), "rb");
+    if (file == nullptr) {
+      return;
+    }
+    fseek(file, 0, SEEK_END);
+    long sz = ftell(file);
+    if (sz == -1) {
+      return;
+    }
+    fseek(file, 0, SEEK_SET);
+
+    _image_release_file_content = (char*) os::malloc(sz + 1, mtInternal);
```

Comment:
For race concerns do everything to a tmp variable and finally do a release_store into `_image_release_file_content`.